cli/src/cmd/forge/fourbyte.rs

```diff
@@ -0,0 +1,119 @@
+use crate::{
+    cmd::forge::build::{CoreBuildArgs, ProjectPathsArgs},
+    compile,
+    opts::forge::CompilerArgs,
+};
+use clap::Parser;
+use ethers::prelude::artifacts::{output_selection::ContractOutputSelection, LosslessAbi};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use tracing::trace;
+
+#[derive(Serialize, Debug)]
+struct ImportRequest {
+    #[serde(rename = "type")]
+    import_type: String,
+    data: Vec<LosslessAbi>,
+}
+
+#[derive(Deserialize, Debug)]
+struct ImportTypeData {
+    imported: HashMap<String, String>,
+    duplicated: HashMap<String, String>,
+}
+
+#[derive(Deserialize, Debug)]
+struct ImportData {
+    function: ImportTypeData,
+    event: ImportTypeData,
+}
+
+#[derive(Deserialize, Debug)]
+struct ImportResponse {
+    result: ImportData,
+}
+
+#[derive(Debug, Clone, Parser)]
+pub struct UploadSelectorsArgs {
+    #[clap(help = "The name of the contract to upload selectors for.")]
+    pub contract: String,
+
+    #[clap(flatten, next_help_heading = "PROJECT OPTIONS")]
+    pub project_paths: ProjectPathsArgs,
+}
+
+impl UploadSelectorsArgs {
+    /// Builds a contract and uploads the ABI to selector database
+    pub async fn run(self) -> eyre::Result<()> {
+        let UploadSelectorsArgs { contract, project_paths } = self;
+
+        let build_args = CoreBuildArgs {
+            project_paths: project_paths.clone(),
+            out_path: Default::default(),
+            ignored_error_codes: vec![],
+            no_auto_detect: false,
+            use_solc: None,
+            offline: false,
+            force: false,
+            libraries: vec![],
+            via_ir: false,
+            revert_strings: None,
+            compiler: CompilerArgs {
+                extra_output: vec![ContractOutputSelection::Abi],
+                ..Default::default()
+            },
+        };
```

Comment:
given that we use this in a couple of places, a nice follow up would be to impl default for that struct so we can do

```rust
CoreBuildArgs {
  ..Default::default()
}
```

perhaps that's even possible already

Reply:
yep good idea - done